packages/lib-classifier/src/hooks/usePanoptesAuth.js

```diff
@@ -1,17 +1,18 @@
-import { useEffect, useState } from 'react'
+import { useState } from 'react'
 
 import { getBearerToken } from '@store/utils'
 
-export default function usePanoptesAuth({ authClient, userID }) {
+export default function usePanoptesAuth({ authClient }) {
   const [authorization, setAuthorization] = useState()
+
   async function checkAuth() {
     const token = await getBearerToken(authClient)
-    setAuthorization(token)
+    if (token !== authorization) {
+      setAuthorization(token)
+    }
   }
 
-  useEffect(function onUserChange() {
-    checkAuth()
-  }, [authClient, userID])
```

Comment:
The original bug happens because this hook stores your bearer token in component state and reuses it forever, even after it has expired. It should be checked (by running `checkAuth()`) every time that it's used.

Reply:
General tip for reviewing React components: whenever you see `useEffect` it has probably introduced a subtle bug.